### PR TITLE
Update changelog for v0.11

### PR DESCRIPTION
**Pull request by radhikaj**
Signed-off-by: Radhika Jandhyala <radhikaj@microsoft.com>

@shnwc  @yentsanglee  This is missing attestation changes in changelog. Can you please suggest what you would like to be added?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Open Enclave SDK Release packages can now be built on non-SGX and non-FLC machines.
 - Support for arbitrarily large thread-local data for SGX machines.
+- Experimental support for OpenSSL inside enclaves has been added while building the SDK from source.
+     - Use BUILD_OPENSSL flag while compiling the SDK
+     - [OpenSSLSupport.md](docs/OpenSSLSupport.md) documents supported options and configuration needed to OpenSSL inside an enclave.
 
 ### Changed
 - Rename the custom claims buffer added by oe_get_evidence from "custom_claims" to "custom_claims_buffer". Likewise, replace the `OE_CLAIM_CUSTOM_CLAIMS` definition for this name with `OE_CLAIM_CUSTOM_CLAIMS_BUFFER`.
+- Building SDK from source		
+      - HAS_QUOTE_PROVIDER cmake option has been removed. This is a continutation of the work in previous release to allo the same build of OE SDK to run on both FLC and non FLC machines."
```
Comment (achamayou) [inline CHANGELOG.md:26]:
```suggestion
      - HAS_QUOTE_PROVIDER cmake option has been removed. This is a continuation of the work in previous release to allow the same build of OE SDK to run on both FLC and non-FLC machines."
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [Unreleased][Unreleased_log]
 --------------
 
+[v0.11.0[[v0.11.0_log]
```
Comment (mingweishih) [inline CHANGELOG.md:13]:
```suggestion
[v0.11.0][v0.11.0_log]
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+--------------
+
 ### Added
-- OE SDK Release packages can now be built on non-SGX and non-FLC machines.
+- Open Enclave SDK Release packages can now be built on non-SGX and non-FLC machines.
```
Comment (mingweishih) [inline CHANGELOG.md:17]:
```suggestion
- Open Enclave SDK release packages can now be built on non-SGX and non-FLC machines.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Support for arbitrarily large thread-local data for SGX machines.
+- Experimental support for OpenSSL inside enclaves has been added while building the SDK from source.
+     - Use BUILD_OPENSSL flag while compiling the SDK
+     - [OpenSSLSupport.md](docs/OpenSSLSupport.md) documents supported options and configuration needed to OpenSSL inside an enclave.
```
Comment (yentsanglee) [inline CHANGELOG.md:22]: Please add these:

- Custom claims buffer serialization/de-serialization helper functions
- `oe_verify_evidence()` and `oe_verify_attestation_certificate_with_evidence()` added to host-verify package
- SGX attestation endorsement claims from `oe_verify_evidence()`
  - `OE_CLAIM_SGX_TCB_INFO`
  - `OE_CLAIM_SGX_TCB_ISSUER_CHAIN`
  - `OE_CLAIM_SGX_PCK_CRL`
  - `OE_CLAIM_SGX_ROOT_CA_CRL`
  - `OE_CLAIM_SGX_CRL_ISSUER_CHAIN`
  - `OE_CLAIM_SGX_QE_ID_INFO`
  - `OE_CLAIM_SGX_QE_ID_ISSUER_CHAIN`

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Changed
 - Rename the custom claims buffer added by oe_get_evidence from "custom_claims" to "custom_claims_buffer". Likewise, replace the `OE_CLAIM_CUSTOM_CLAIMS` definition for this name with `OE_CLAIM_CUSTOM_CLAIMS_BUFFER`.
+- Building SDK from source		
+      - HAS_QUOTE_PROVIDER cmake option has been removed. This is a continutation of the work in previous release to allo the same build of OE SDK to run on both FLC and non FLC machines."
+      - Intel SGX EnclaveCommonAPI pacakges are no longer needed to build the SDK.
```
Comment (yentsanglee) [inline CHANGELOG.md:28]: Please add these to "Changed" list:

- `oe_verify_attestation_certificate_with_evidence()` can verify certificates generated by `oe_generate_attestation_certificate()` as well as `oe_get_attestation_certificate_with_evidence()`.
- SGX attestation evidence internal structure changed. Current structure (version 3) is not compatible with previous version. Applications that call `oe_get_evidence()` or `oe_verify_evidence()` have to be rebuilt.
- Some SGX attestation format IDs are renamed:

| Old | New |
| -- | - |
OE_FORMAT_UUID_SGX_ECDSA_P256  | OE_FORMAT_UUID_SGX_ECDSA |
OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT | OE_FORMAT_UUID_LEGACY_REPORT_REMOTE |
OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE | OE_FORMAT_UUID_RAW_SGX_QUOTE_ECDSA |

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Removed
+- COMPILE_SYSTEM_EDL has been removed.
```
Comment (yentsanglee) [inline CHANGELOG.md:31]: Please add this to "Remove" list:

- The following SGX attestation format ID is not yet availble:
  - `OE_FORMAT_UUID_SGX_ECDSA_P384`

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,14 +10,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Support for arbitrarily large thread-local data for SGX machines.
+- Experimental support for OpenSSL inside enclaves has been added while building the SDK from source.
+     - Use BUILD_OPENSSL flag while compiling the SDK
+     - [OpenSSLSupport.md](docs/OpenSSLSupport.md) documents supported options and configuration needed to OpenSSL inside an enclave.
 
 ### Changed
 - Rename the custom claims buffer added by oe_get_evidence from "custom_claims" to "custom_claims_buffer". Likewise, replace the `OE_CLAIM_CUSTOM_CLAIMS` definition for this name with `OE_CLAIM_CUSTOM_CLAIMS_BUFFER`.
+- Building SDK from source		
+      - HAS_QUOTE_PROVIDER cmake option has been removed. This is a continuation of the work in previous release to allow the same build of OE SDK to run on both FLC and non-FLC machines."
+      - Intel SGX EnclaveCommonAPI pacakges are no longer needed to build the SDK.
```
Comment (qiucwang) [inline CHANGELOG.md:28]: Should sample change be reflected in CHANGELOG as well?

- The attestation functions in local_attestation/remote_attestation/attested_tls/host_verify samples are changed. Now they use attestation plugin APIs, defined in `attestation/attester.h and attestation/verifier.h` to generate and verify evidence.

**Review by shnwc — COMMENTED** [review]
Please add this bullet in to "Added" list:
* `oe_get_evidence()` support of generation of SGX EPID evidences, in formats `OE_FORMAT_UUID_SGX_EPID_LINKABLE` and `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE`.

Please add these bullets in the "Removed" list:

* Declaration of SGX format ID `OE_FORMAT_UUID_SGX_ECDSA_P384` is removed
* `oe_get_evidence()` support of SGX legacy formats `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` and `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` is removed.

File: CHANGELOG.md (hunk 7 above)
Comment (radhikaj) [reply CHANGELOG.md:28]: Done.

File: CHANGELOG.md (hunk 4 above)
Comment (radhikaj) [reply CHANGELOG.md:22]: I changed the wording a little. Can you please verify that it is accurate?

**Comment by radhikaj** [comment]
> Please add this bullet in to "Added" list:
> 
> * `oe_get_evidence()` support of generation of SGX EPID evidences, in formats `OE_FORMAT_UUID_SGX_EPID_LINKABLE` and `OE_FORMAT_UUID_SGX_EPID_UNLINKABLE`.
> 
> Please add these bullets in the "Removed" list:
> 
> * Declaration of SGX format ID `OE_FORMAT_UUID_SGX_ECDSA_P384` is removed
> * `oe_get_evidence()` support of SGX legacy formats `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` and `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` is removed.

Thanks @shnwc, done.

File: CHANGELOG.md [hunk 8]
```diff
@@ -10,15 +10,50 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
 ### Added
-- OE SDK Release packages can now be built on non-SGX and non-FLC machines.
+- Open Enclave SDK release packages can now be built on non-SGX and non-FLC machines.
 - Support for arbitrarily large thread-local data for SGX machines.
+- Experimental support for OpenSSL inside enclaves has been added while building the SDK from source.
+     - Use BUILD_OPENSSL flag while compiling the SDK.
+     - [OpenSSLSupport.md](docs/OpenSSLSupport.md) documents supported options and configuration needed to use OpenSSL inside an enclave.
+- Custom claims buffer serialization/de-serialization helper functions
+oe_verify_evidence() and oe_verify_attestation_certificate_with_evidence() have been added to the host-verify package.
```
Comment (yentsanglee) [inline CHANGELOG.md:23]: This is a bullet by itself:

**-** oe_verify_evidence() and oe_verify_attestation_certificate_with_evidence() have been added to the host-verify package.

**Review by qiucwang — APPROVED** [review]
LGTM

**Review by yentsanglee — APPROVED** [review]

**Comment by radhikaj** [comment]
Thanks everybody for helping fix this up.

**Pull request merged**